mindmeld/models/nn_utils/input_encoders.py

```diff
@@ -0,0 +1,937 @@
+            _inputs = self._bert_tokenizer(
+                examples, padding=True, truncation=True, max_length=padding_length,
+                return_tensors="pt"
+            ).to(self.device)
+            _outputs = self._bert_model(**_inputs, return_dict=True)
+
+            # last_hidden_state, pooler_output, seq_lengths are the keys being updated
+            return_dict.update({
+                **_outputs,
+                "seq_lengths": _inputs["attention_mask"].sum(dim=-1)
+            })
+
+        if labels:
+            return_dict.update({"labels": torch.as_tensor(labels, dtype=torch.long)})
+
+        return return_dict
+
+
+class TokenClsEncoderWithPlmLayer(PretrainedLangModelBasedEncoder):
+
+    def _reformat_and_validate(self, examples, labels=None):
+        super()._reformat_and_validate(examples, labels)
+
+        # validation for labels' lengths too
+        if labels:
+            for ex, label_tokens in zip(examples, labels):
+                ex_tokens = ex.split(" ")
+                if len(ex_tokens) != len(label_tokens):
+                    msg = f"Number of tokens in a sentence ({len(ex_tokens)}) must be same as the" \
+                          f"number of tokens in the corresponding token labels " \
+                          f"({len(label_tokens)}) for sentence '{ex}' with labels '{labels}'"
+                    raise AssertionError(msg)
+
+        return examples, labels
+
+    @staticmethod
+    def _trim_combined(x: List[List[Any]], trim_len: int, y: List[Any] = None):
+        curr_len = 0
+        if y:
+            new_x, new_y = [], []
+            for _x, _y in zip(x, y):
+                if curr_len >= trim_len:
+                    return new_x, new_y
+                if curr_len + len(_x) > trim_len:
+                    new_x.append(_x[trim_len - curr_len])
+                    new_y.append(_y)
+                elif curr_len + len(_x) <= trim_len:
+                    new_x.append(_x)
+                    new_y.append(_y)
+            return new_x, new_y
+        else:
+            new_x = []
+            for _x in x:
+                if curr_len >= trim_len:
+                    return new_x
+                if curr_len + len(_x) > trim_len:
+                    new_x.append(_x[trim_len - curr_len])
+                elif curr_len + len(_x) <= trim_len:
+                    new_x.append(_x)
+            return new_x
+
+    def batch_encode(
+        self,
+        examples: Union[str, List[str]],
+        labels: Union[int, List[int]] = None,
+        padding_length: int = None,
+    ):
+        examples, labels = self._reformat_and_validate(examples, labels)
+
+        return_dict = {}
+        padding_length = padding_length or self.padding_length
+
+        if self.embedder_type == "bert":
+            # tokenize each word of each input seperately
+            tokenized_examples = [
+                [self._bert_tokenizer.tokenize(word) for word in example.split(" ")]
```

Comment:
why do we use whitespace splitting here?

Reply:
modified and the assumption is not included anymore